x-pack/plugins/timelines/public/components/t_grid/toolbar/fields_browser/field_table.tsx

```diff
@@ -103,6 +108,33 @@ const FieldTableComponent: React.FC<FieldTableProps> = ({
     [columnHeaders, dispatch, timelineId]
   );
 
+  /**
+   * Pagination controls
+   */
+  const pagination: Pagination = useMemo(
+    () => ({
+      pageIndex,
+      pageSize,
+      totalItemCount: fieldItems.length,
+      pageSizeOptions: [10, 25, 50],
+    }),
+    [fieldItems.length, pageIndex, pageSize]
+  );
+
+  const onTableChange = useCallback(({ page }: CriteriaWithPagination<BrowserFieldItem>) => {
+    const { index, size } = page;
+    setPageIndex(index);
+    setPageSize(size);
+  }, []);
+
+  useEffect(() => {
+    // Resets the pagination when some filter has changed, consequently, the number of fields is different
+    setPageIndex(0);
+  }, [fieldItems.length]);
```

Comment:
I'm just wondering what will happen if the result length is the same for a different filter. This will not trigger this effect.

Reply:
Makes sense, thanks for explaining how much of a corner-case this actually is!